plugins/org.jkiss.dbeaver.ext.clickhouse/src/org/jkiss/dbeaver/ext/clickhouse/model/ClickhouseSQLDialect.java

```diff
@@ -104,4 +109,10 @@ public String getColumnTypeModifiers(@NotNull DBPDataSource dataSource, @NotNull
     public boolean supportsNestedComments() {
         return true;
     }
+
+    //We should quote keywords which is not keywords for clickhouse, otherwise JSQLParser can't parse statements
+    @Override
+    public boolean mustBeQuoted(@NotNull String str, boolean forceCaseSensitive) {
+        return Arrays.stream(CLICKHOUSE_NONKEYWORDS).anyMatch(k -> k.equalsIgnoreCase(str)) || super.mustBeQuoted(str, forceCaseSensitive);
```

Comment:
I don't like the capturing lambda here. Let's unwrap the stream into a plain loop.